lib/fluent/plugin/filter_bwlist.rb

```diff
@@ -15,12 +15,52 @@
 
 require "fluent/plugin/filter"
 
+require 'aws-sdk-s3'
+
 module Fluent
   module Plugin
-    class BwlistFilter < Fluent::Plugin::Filter
+    class BWlistFilter < Fluent::Plugin::Filter
       Fluent::Plugin.register_filter("bwlist", self)
 
+      config_param :key, :string,
+                   desc: "target key"
+
+      config_param :s3_bucket, :string,
+                   desc: 'config bucket'
+
+      config_param :s3_key, :string,
+                   desc: 'config key'
+
+      config_param :mode, :enum, list: [:black, :white], default: :black,
+                   desc: 'select mode black/white'
+
+      def configure(conf)
+        super
+
+        @mode_bool = conf['mode'] == 'black'
+      end
+
+      def start
+        super
+
+        @list = fetch_list(@s3_bucket, @s3_key)
+        log.info "filter_bwlist:: s3:#{@s3_bucket}/#{@s3_key} key:#{@key} list_size:#{@list.length}"
+        log.debug "filter_bwlist:: list:#{@list}"
+      end
+
       def filter(tag, time, record)
+        @list.include?(record[@key].to_s) ^ @mode_bool ? record : nil
+      rescue => e
```

Comment:
[FYI] You don't need to rescue errors. If an error occurs in filter plugin, `emit_error_event` will be called (cf. [Fluent::Plugin::Filter#filter_stream](https://github.com/fluent/fluentd/blob/v1.3.1/lib/fluent/plugin/filter.rb#L56),  [Fluent::EventRouter::Pipeline::FilterOptimizer#optimized_filter_stream](https://github.com/fluent/fluentd/blob/v1.3.1/lib/fluent/event_router.rb#L191-L205)). We can capture it by setting `<label @ERROR>`.
cf. https://docs.fluentd.org/v1.0/articles/config-file#@error-label

`log.warn` seems to be called even if we don't set `<label @ERROR>`.
cf. https://github.com/fluent/fluentd/blob/v1.3.1/lib/fluent/root_agent.rb#L310-L311